helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/OptimalAssignment.java

```diff
@@ -88,8 +152,29 @@ public boolean hasAnyFailure() {
     return !_failedAssignments.isEmpty();
   }
 
-  public String getFailures() {
-    // TODO: format the error string
-    return _failedAssignments.toString();
+  public String getErrorMessage() {
```

Comment:
1. I'm afraid this might pollute the log big time.
2. What value do we have in printing out all failed assignments? I thought as soon as we encounter a failed assignment, we just terminate? So what we'll see here is always just the first failed assignment?